pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/module/VanityVariantJsonServlet.java

```diff
@@ -0,0 +1,143 @@
+                Constants.SERVICE_VENDOR + "=Red Hat Content Tooling team"
+        })
+@SlingServletPaths(value = "/api/module/vanity")
+// /api/module/vanity.json/{locale}/{productLabel}/{versionLabel}/{vanityUrl}
+public class VanityVariantJsonServlet extends SlingSafeMethodsServlet {
+
+    private final SlingPathSuffix suffix = new SlingPathSuffix("/{locale}/{productLabel}/{versionLabel}/{vanityUrl}");
+
+    @Override
+    protected void doGet(@NotNull SlingHttpServletRequest request, @NotNull SlingHttpServletResponse response) throws ServletException, IOException {
+        Map<String, String> params = suffix.getParameters(request);
+        String locale = params.get("locale");
+        String productLabel = params.get("productLabel");
+        String versionLabel = params.get("versionLabel");
+        String vanityUrl = params.get("vanityUrl");
+
+        JcrQueryHelper queryHelper = new JcrQueryHelper(request.getResourceResolver());
+        try {
+
+            // Find the metadata with the right vanity url,
+            Optional<ModuleVersion> moduleVersion = queryHelper.query(
+                    "select * from [pant:moduleVersion] as v where v.[metadata/urlFragment] = '/" + vanityUrl + "'")
+                    .map(resource -> SlingModels.getModel(resource, ModuleVersion.class))
+                    // the right locale,
+                    .filter(modVer -> {
+                        String normalizedLocaleCode = ULocale.canonicalize(locale);
+                        String normalizedModuleLocaleCode = ULocale.canonicalize(modVer.getParent().getParentLocale().getName());
+                        return normalizedLocaleCode.equals(normalizedModuleLocaleCode);
+                    })
+                    // the right version,
+                    .filter(modVer -> {
+                        try {
+                            return modVer.metadata().get().productVersion().getReference().name().get().equals(versionLabel);
+                        } catch (RepositoryException e) {
+                            throw new RuntimeException(e);
+                        }
+                    })
+                    // and the right product
+                    .filter(modVer -> {
+                        try {
+                            return modVer.metadata().get().productVersion().getReference().getProduct().urlFragment().get().equals(productLabel);
+                        } catch (RepositoryException e) {
+                            throw new RuntimeException(e);
+                        }
+                    })
+                    // There should be 1 at most, but get the first if there are more
+                    .findFirst();
+
+            if(!moduleVersion.isPresent()) {
+                // try to find by variant uuid, instead of vanity url
+                moduleVersion = findModuleByUuid(queryHelper, productLabel, versionLabel, locale, vanityUrl);
```

Comment:
I'm wondering why this "vanity" servlet is also accepting the UUID - perhaps you envision this new servlet as "the" one-and-only servlet that Drupal would call, and it handles anything you throw at it? Can you clarify if that's what you intended?